src/lib.rs

```diff
@@ -286,30 +279,21 @@ impl Default for RequestParams {
 }
 
 /// Get a default `Client` and `RequestParams`.
-pub fn default() -> (hyper::Client, RequestParams) {
-    (hyper::Client::new(), RequestParams::default())
-}
-
-macro_rules! req_with_body {
-    ($client:ident, $url:ident, $req:ident, $params:ident, $method:ident) => ({
-    	let body = $req.body.expect("Expected this request to have a body. This is a bug, please file an issue on GitHub.");
-
-		let body: &[u8] = (*body).borrow();
-		let mut cursor = Cursor::new(body);
-
-		$client.$method(&$url).headers($params.headers.to_owned()).body(&mut cursor).send()
-	})
+pub fn default() -> (reqwest::Client, RequestParams) {
+    //FIXME: Bad in libs
+    let client = reqwest::Client::new().unwrap();
```

Comment:
It was a bit strange at first, especially as you seemed to pull in `hyper` in examples, etc regardless in order to customize the requests.

I would say keep it, and return  `Result<(reqwest::Client, RequestParams)>`.

Quick side note, it took me a while to grok that the elasticsearch url is in `RequestParams`, an example or note of it front and centre would be nice - its a common thing to change. (The test `request_params_has_default_base_url` at least shows where)